src/BinaryPack/Serialization/Processors/ObjectProcessor{T}.cs

```diff
@@ -256,9 +257,26 @@ protected override void EmitDeserializer(ILGenerator il)
                 il.Emit(OpCodes.Ret);
                 il.MarkLabel(isNotNull);
 
+                ConstructorInfo? parameterlessConstructor = typeof(T).GetConstructor(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance, null, Type.EmptyTypes, null);
+
+                if (parameterlessConstructor == null)
+                {
+                    ConstructorInfo? parameterConstructor = typeof(T).GetConstructor(Members.Select(member => member.GetMemberType()).ToArray());
+
+                    if (parameterConstructor == null)
+                    {
+                        throw new NullReferenceException("The given object has neither an parameterless constructor nor a constructor with parameters named like all serialized fields/properties.");
```

Comment:
A better exception type should be used here (see the comment in TypeExtensions.cs)